utils/evaluate_utils.py

```diff
@@ -13,12 +13,83 @@
 
 from train_utils import get_X_y_data
 
+from sklearn.metrics import precision_recall_curve
+from sklearn.preprocessing import label_binarize
+
 # set numpy seed to make random operations reproduceable
 np.random.seed(0)
 
-def evaluate_model_cm(
-    log_reg_model: LogisticRegression, dataset: pd.DataFrame
-):
+
+def class_PR_curves(single_cell_data: pd.DataFrame, log_reg_model):
+    """
+    save precision recall curves for each class to the save directory
+    also, return the precision/recall data for each class in tidy long format
+
+    Parameters
+    ----------
+    single_cell_data : pd.DataFrame
+        single cell dataframe with correct cell metadata and feature data
+    log_reg_model : sklearn classifier
+        clasifier to get precision recall curves for
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with precision/recall data for each class in tidy long format
+
+    matplotlib.figure.Figure
+        figure with compilation of all class PR curves
+    """
+
+    phenotypic_classes = log_reg_model.classes_
+    X, y = get_X_y_data(single_cell_data)
+
+    # binarize labels for precision recall curve function
+    y_binarized = label_binarize(y, classes=phenotypic_classes)
+    # predict class probabilities for feature data
+    y_score = log_reg_model.predict_proba(X)
+
+    # data from PR curves will be stored in tidy data format (eventually pandas dataframe)
+    PR_data = []
+    # which thresholds the precision/recalls correspond to
+    PR_threshold = np.arange(single_cell_data.shape[0])
+    # last values in precision/recall curve don't correspond to cell dataset
+    PR_threshold = np.append(PR_threshold, None)
+
+    fig, axs = plt.subplots(4, 4)
+    fig.set_size_inches(15, 15)
+    ax_x = 0
+    ax_y = 0
+    for i in range(phenotypic_classes.shape[0]):
+        precision, recall, _ = precision_recall_curve(y_binarized[:, i], y_score[:, i])
+        class_tidy_data = pd.DataFrame(
+            {
+                "Phenotypic_Class": phenotypic_classes[i],
+                "PR_Threshold": PR_threshold,
+                "Precision": precision,
+                "Recall": recall,
+            }
+        )
+        PR_data.append(class_tidy_data)
```

Comment:
If the variable `class_tidy_data` doesn't get used again within this loop after being appended to `PR_data`, think about adding the `pd.DataFrame` directly to your append call (without a variable assigned). The difference here might be minor, and is completely optional, but could save a line or two of code.